src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerOptions.cs

```diff
@@ -118,5 +118,21 @@ public JwtBearerOptions() : base()
         /// <see cref="Http.Authentication.AuthenticationProperties"/> after a successful authorization.
         /// </summary>
         public bool SaveToken { get; set; } = true;
+
+        /// <summary>
+        /// Authentication failures with one of these exception types, or types derived from these, will have their messages
+        /// included in a 'X-Authentication-Failure' response header.
+        /// </summary>
+        public ICollection<Type> ReportableFailures { get; set; } = new List<Type>()
```

Comment:
On the other hand, we really need an option to disable this feature when needed.